### ISSUE TITLE
incorrect assigning-non-slot when __slots__ is set up from constants

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#379]
Originally reported by: **Anonymous**

----------------------------------------

using pylint-1.3.1 with this snippet:

```
#!python
class RunAttributes(object):
    REGULAR_ATTRS = frozenset(('a', 'b',))
    __slots__ = tuple(REGULAR_ATTRS) + ('_foo', '_bar')

    def __init__(self):
        self._foo = 1
        self.a = None
```

pylint incorrectly complains:
```
E:  6, 8: Assigning to attribute '_foo' not defined in class slots (assigning-non-slot)
E:  7, 8: Assigning to attribute 'a' not defined in class slots (assigning-non-slot)
```

dropping the frozenset/tuple logic lets it pass.  but i don't want to do that :).

----------------------------------------
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/380

**Comment by PCManticore** [comment]
This should be fixed in astroid 2.0 with https://github.com/PyCQA/astroid/commit/3f6af118d19342fc92167c78865f58461e091ccc. Unfortunately, astroid 2.0 doesn't work yet with pylint, but you can test this by running this sample code, where you can see that .slots() is properly inferred now.

    from astroid.test_utils import extract_node
    n = extract_node('''
    class Issue(object):
      SlotDefaults = {'id': 0}
      __slots__ = SlotDefaults.keys()

    ''')
    n = next(n.infer())
    print(n)
    print(n.slots())
